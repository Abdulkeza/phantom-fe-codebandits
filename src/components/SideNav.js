import React from 'react';
import { useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import dashIcon from '../images/dashboardIcon.png';
import manageIcon from '../images/manageIcon.png';
import notificationIcon from '../images/notificationIcon.png';
import profileIcon from '../images/profileIcon.png';
import settingIcon from '../images/settingIcon.png';
import supportIcon from '../images/supportIcon.png';
import profileImg from '../images/profileImg.jpg';
import { ButtonA as Button } from './Button.js';
import SideNavLink from './SideNavLink.js';

const SideNav = ({ styles }) => {
  const navigate = useNavigate();
  const logout = () => {
    toast('You are logged out', { type: 'success' });
    navigate('/login');
  };

  return (
    <main className={`${styles} flex text-sm`}>
      <aside className="border-r-2 h-full sticky top-0 px-8 py-4 bg-gray-100 transition-transform">
        <div className=" mb-6">
          <span className="rounded-full mx-auto flex items-center justify-center h-12 w-12 mb-5">
            <img
              src={profileImg}
              alt="Image"
              className="rounded-full object-cover h-12 w-12"
            />
          </span>
          <div className="text-center font-black text-md">
            <h1>NGABO Patrick</h1>
            <div>
              <h1 className="font-bold">Operator</h1>
            </div>
          </div>
        </div>
        <div>
          <h2 className="mb-2 text-lg font-black">Account</h2>
          <ul className="mb-2">
            <SideNavLink image={dashIcon} linkTo="/account" name="Dashboard" />
            <SideNavLink image={manageIcon} linkTo="#p" name="Management" />
            <SideNavLink
              image={profileIcon}
              linkTo="/dashboard/profile"
              name="Profile"
            />
            <SideNavLink
              image={notificationIcon}
              linkTo="#p"
              name="Notifications"
            />
            <SideNavLink image={supportIcon} linkTo="#p" name="Support" />
            <SideNavLink image={settingIcon} linkTo="#p" name="Settings" />
          </ul>
          <Button
            name="Logout"
            type="submit"
            id="logout"
            styles="bg-red px-6 py-1 hover:bg-red-600"
            onClick={logout}
          />
        </div>
      </aside>
    </main>
  );
};

export default SideNav;
