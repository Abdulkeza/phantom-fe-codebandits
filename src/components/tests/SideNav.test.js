import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import SideNav from '../SideNav.js';

describe('SideNav', () => {
  it('should render the SideNav component', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <SideNav />)
        </MemoryRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
