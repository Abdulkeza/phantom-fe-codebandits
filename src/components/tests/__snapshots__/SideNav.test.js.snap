// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render the SideNav component 1`] = `
<main
  className="undefined flex text-sm"
>
  <aside
    className="border-r-2 h-full sticky top-0 px-8 py-4 bg-gray-100 transition-transform"
  >
    <div
      className=" mb-6"
    >
      <span
        className="rounded-full mx-auto flex items-center justify-center h-12 w-12 mb-5"
      >
        <img
          alt="Image"
          className="rounded-full object-cover h-12 w-12"
          src="profileImg.jpg"
        />
      </span>
      <div
        className="text-center font-black text-md"
      >
        <h1>
          NGABO Patrick
        </h1>
        <div>
          <h1
            className="font-bold"
          >
            Operator
          </h1>
        </div>
      </div>
    </div>
    <div>
      <h2
        className="mb-2 text-lg font-black"
      >
        Account
      </h2>
      <ul
        className="mb-2"
      >
        <div
          className="flex justify-center items-center hover:bg-gray-200 py-1"
        >
          <img
            alt="icon"
            className="w-6"
            src="dashboardIcon.png"
          />
          <a
            className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
            href="/dashboard/main"
            onClick={[Function]}
          >
            <li>
              Dashboard
            </li>
          </a>
        </div>
        <div
          className="flex justify-center items-center hover:bg-gray-200 py-1"
        >
          <img
            alt="icon"
            className="w-6"
            src="manageIcon.png"
          />
          <a
            className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
            href="/management"
            onClick={[Function]}
          >
            <li>
              Management
            </li>
          </a>
        </div>
        <div
          className="flex justify-center items-center hover:bg-gray-200 py-1"
        >
          <img
            alt="icon"
            className="w-6"
            src="profileIcon.png"
          />
          <a
            className="py-1 px-2 font-bold my-1  cursor-pointer w-full"
            href="/profile"
            onClick={[Function]}
          >
            <li>
              Profile
            </li>
          </a>
        </div>
        <div
          className="flex justify-center items-center hover:bg-gray-200 py-1"
        >
          <img
            alt="icon"
            className="w-6"
            src="notificationIcon.png"
          />
          <a
            aria-current="page"
            className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
            href="/#p"
            onClick={[Function]}
          >
            <li>
              Notifications
            </li>
          </a>
        </div>
        <div
          className="flex justify-center items-center hover:bg-gray-200 py-1"
        >
          <img
            alt="icon"
            className="w-6"
            src="supportIcon.png"
          />
          <a
            aria-current="page"
            className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
            href="/#p"
            onClick={[Function]}
          >
            <li>
              Support
            </li>
          </a>
        </div>
        <div
          className="flex justify-center items-center hover:bg-gray-200 py-1"
        >
          <img
            alt="icon"
            className="w-6"
            src="settingIcon.png"
          />
          <a
            aria-current="page"
            className="py-1 px-2 font-bold my-1  cursor-pointer w-full active"
            href="/#p"
            onClick={[Function]}
          >
            <li>
              Settings
            </li>
          </a>
        </div>
      </ul>
      <div>
        <button
          className="bg-red px-6 py-1 hover:bg-red-600 bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
          onClick={[Function]}
          type="submit"
        >
          Logout
        </button>
      </div>
    </div>
  </aside>
</main>
`;
